eta/core/events.py

```diff
@@ -27,7 +31,194 @@
 import numpy as np
 
 from eta.core.config import Config, Configurable
-from eta.core.serial import Serializable
+from eta.core.data import AttributeContainer
+from eta.core.serial import Container, Serializable
+
+
+class DetectedEvent(Serializable):
+    '''A detected event in a video.
+
+    Attributes:
+        label: the event label
+        event_series: the EventSeries (discontiguous temporal) range of the
+            event
+        confidence: (optional) the detection confidence in [0, 1]
+        index: (optional) an index assigned to the event
+        score: (optional) a score assigned to the event
+        attrs: (optional) an AttributeContainer describing additional
+            attributes of the event
+    '''
+
+    def __init__(self, label, event_series, confidence=None, index=None,
+                 score=None, attrs=None):
+        '''Creates a DetectedEvent instance.
+
+        Args:
+            label: the event label
+            event_series: the EventSeries (discontiguous temporal) range of the
+                event
+            confidence: (optional) the detection confidence in [0, 1]
+            index: (optional) an index assigned to the event
+            score: (optional) a score assigned to the event
+            attrs: (optional) an AttributeContainer describing additional
+                attributes of the event
+        '''
+        self.label = label
+        self.event_series = event_series
```

Comment:
I'm super motivated to replace `EventSeries` with `eta.core.video.FrameRanges` so that we can rename this field to `frames`.

A bit confusing that an "event" contains an "event_series" field.

I plan to make a PR into this that makes that tweak

Reply:
done